astropy/table/tests/test_table.py

```diff
@@ -1390,6 +1391,22 @@ def test_byteswap_fits_array(self, table_types):
                 assert (data[colname].dtype.byteorder
                         == arr2[colname].dtype.byteorder)
 
+    def test_convert_numpy_object_array(self, table_types):
+        d = table_types.Table([[1, 2], [3, 4]], names=('a', 'b'))
+
+        # Single table
+        np_d = np.array(d, dtype=object)
+        assert isinstance(np_d, np.ndarray)
+        assert np_d[()] is d
+
+    @pytest.mark.xfail(NUMPY_LT_1_20, reason="numpy array introspection changed")
+    def test_convert_list_numpy_object_array(self, table_types):
+        d = table_types.Table([[1, 2], [3, 4]], names=('a', 'b'))
+        ds = [d, d, d]
```

Comment:
Might be a bit paranoid, but perhaps one of the tables should have different values to make sure it is not taking 3 times the first object in the list.

Reply:
That seems a bit paranoid to test make sure that `np.array([x, y, z], dtype=object)` actually puts `y` in the 2nd element instead of `x`. I'm pretty confident this bug does not exist in numpy at this point.

Reply:
Please open a new issue if this needs follow-up. Thanks!